pandas/_libs/tslibs/offsets.pyx

```diff
@@ -212,13 +212,66 @@ def _validate_business_time(t_input):
             raise ValueError("time data must match '%H:%M' format")
     elif isinstance(t_input, dt_time):
         if t_input.second != 0 or t_input.microsecond != 0:
-            raise ValueError(
-                "time data must be specified only with hour and minute")
+            raise ValueError("time data must be specified only with "
+                             "hour and minute")
         return t_input
     else:
         raise ValueError("time data must be string or datetime.time")
 
 
+def _validate_n(n):
+    """
+    Require that `n` be a nonzero integer.
+
+    Parameters
+    ----------
+    n : int
+
+    Raises
+    ------
+    ValueError
+    """
+    if n == 0 or not isinstance(n, int):
+        raise ValueError("N cannot be 0")
+
+
+def _validate_month(month):
+    """
+    Require that `month` be an integer between 1 and 12 inclusive.
+
+    Parameters
+    ----------
+    month : int
```

Comment:
I would guess we have this type of method in multiple places (though I agree these should actually be in this module).